packages/mobx/src/types/observableannotation.ts

```diff
@@ -38,6 +46,62 @@ function extend_(
     )
 }
 
+function decorate_20223_(
+    this: Annotation,
+    desc,
+    context: ClassAccessorDecoratorContext | ClassFieldDecoratorContext
+) {
+    assert20223DecoratorType(context, ["accessor", "field"])
+
+    const ann = this
+    const { kind, name: key, addInitializer } = context
+
+    // Backwards/Legacy behavior
+    if (kind == "field") {
+        addInitializer(function () {
+            const adm: ObservableObjectAdministration = asObservableObject(this)[$mobx]
+            adm.defineObservableProperty_(
+                key,
+                (this as any)[key],
+                ann.options_?.enhancer ?? deepEnhancer,
+                false
+            )
+        })
+        return
+    }
+
+    // For the most backwards compatibility, this will ensure that the observable is an enumerable "own" prop.
+    //   However, since the user has to opt-in to new logic (via the `accessor` keyword), it may be nice _not_ to do so since
+    //   that's kind of contrary tot he direction that ES is going.
+    addInitializer(function () {
+        Object.defineProperty(this, key, {
+            ...getCachedFallthroughPropDescriptor(key),
```

Comment:
Likely - due to the aforementioned gotcha with when `addInitializer`s run, I think I'm going to need to tweak these a good bit.
It's looking like I may need to make the `adm.values_.set(key, observable)` (and leading lines) lazy.

Given
```typescript
function Dec1(acc, context) {
  return {
    get() { },
    set() { },
    init() { }
  }
}

class Thing {
  @Dec1
  @Dec2
  accessor x = 5
}
```
The process upon instantiation goes like so:
- Run `addInitializer` callbacks
- Call `Dec2.init(5)`
- Call `Dec1.init(<result of ^>)` (repeating from inner-most to outer-most decorator.init)
- Set Private field value to ^
Neither Dec1 nor Dec2 receive `set` upon initialization, and if I use an `addInitializer`, it has 0 access to that initial value 5... (If I try, the runtime blows up thinking that I tried to inappropriately access a `#private_slot`)

I'm going to look more deeply Monday and see determine the best approach. Lazy initialization isn't as ideal performance wise since it has to check if it needs to be initialized, but (w/o requiring the continued use of `makeObbservable()`), it's the only way (I've found so far) to reliably have access to everything that's needed.

Reply:
FWIW we never supported multiple decorators on a single member in MobX (it might or might not work, no guarantees). So feel free to go with the simplest / fastest approach, rather than the most correct one for now.

Reply:
More for third-party decorators - not using multiple MobX decorators. Definitely a rarer use case.

Reply:
sorry, yeah was referring to third-party decorators. We don't have meaningful combinations of decorators in MobX itself.

Reply:
:+1: